quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_format.rs

```diff
@@ -85,10 +85,16 @@ impl FromStr for DateTimeFormat {
             "unix_ts_millis" => DateTimeFormat::TimestampMillis,
             "unix_ts_micros" => DateTimeFormat::TimestampMicros,
             "unix_ts_nanos" => DateTimeFormat::TimestampNanos,
-            _ => DateTimeFormat::Strftime {
-                strftime_format: date_time_format_str.to_string(),
-                with_timezone: date_time_format_str.contains("%z"),
-            },
+            _ => {
+                // Validate the format.
+                let _ = parse_to_format_item(date_time_format_str).map_err(|err| {
```

Comment:
Arg, the whole point of switching from `chrono` to `time` was to be able to hold `FormatItem` in `DateTimeFormat::Strftime` so that we don't have to parse the time format for each date parsed.

I'm realizing now that `FormatItem<'a>` carries a lifetime, and that's not possible... Since we have to keep `chrono` for parsing cron schedule for retention policies, I'm not sure this PR is worth it anymore.

@fulmicoton WDYT?

Reply:
We can overcome this, The current blocker right now is that some parse tree `Vec<FormatItem>` produced by `time-fmt` cannot be used correctly by `time-rs` to parse so I will investigate that here https://github.com/MiSawa/time-fmt/issues/11.